design-proposals/edge-infra-cve-tracking.md

```diff
@@ -0,0 +1,184 @@
+# Design Proposal: CVE Tracking for OS Profile and Edge Nodes
+
+Author(s): Daniele Moro
+
+Last updated: 2025-05-08
+
+## Abstract
+
+This proposal describes a mechanism for tracking CVEs (Common Vulnerabilities
+and Exposures) for OS Profiles and Edge Nodes in the Edge Infrastructure Manager
+(EIM).
+
+## Proposal
+
+Currently, we have no mechanisms to track CVEs related to the OS in Edge
+Infrastructure Manager. Ideally, we should have a mechanism to track that a
+certain version of the OS is affected by a certain CVE, and that a set of Edge
+Nodes are affected too. Also, the user of EIM should be aware of this, and it
+should be provided with information about which version of the OS fixed the
+given CVE. Users of EIM should be also able to see both forward and backward
+list of fixed CVEs. In order to understand the effect on CVEs on updates.
+
+Requirements:
+- Report existing CVEs for a given OS Profile version
+- Report fixed CVEs on a given OS Profile version
+- Report list of CVEs that affect an Instance at runtime
+- Do backward and forward fixed CVEs tracking between OSProfile versions
+
+The proposal is to extend both the OS Profile and the Instance resources to
+contain information about CVEs.
+
+**OS Profile**:
+- `existing_cves`: a list of CVEs that affect the given OS Profile version. This
+  is filled from information provided by the OS vendor.
+- `fixed_cves`: a list of CVEs that was fixed in the given OS Profile Version.
+  This is filled from information provided by the OS vendor.
+
+**Instance**:
+- `cves`: a list of CVEs that affects the given Instance at runtime. The
+  information on this field comes from the Edge Node itself. For Immutable OSes
+  this should contain the same values as the `affected_cves` from the
+  `current_os` OS Profile.
+
+Since we support both Immutable OSes, in particular the Edge Microvisor Toolkit
+(EMT), and mutable OSes, in particular Ubuntu, we need to design support for CVE
+reporting for both OSes. Unfortunately, there is no standard way to publish CVEs
+report, so this feature needs to be designed and implemented on a per-OS basis.
+
+### OS Profile
+
+For the OS profile, for both `affected_cves` and `fixed_cves` we expect support
+from the OS Vendor or Provider to provide these information.
+
```

Comment:
I'd maybe go the direction of defining the model that we want internally for the CVEs (see comment at the beginning) and then mention here that we will import CVes from different sources into that model.